src/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/RedirectHandler.cs

```diff
@@ -54,10 +54,20 @@ public RedirectHandler(int maxAutomaticRedirections, HttpMessageHandler initialI
                 // Clear the authorization header.
                 request.Headers.Authorization = null;
 
+                if (NetEventSource.IsEnabled)
+                {
+                    NetEventSource.Info(this, $"Redirecting from {request.RequestUri} to {redirectUri} in response to status code {(int)response.StatusCode} '{response.StatusCode}'.");
```

Comment:
I'm a little concerned we're starting to become inconsistent (or maybe we already were and this just makes it more so) inconsistent on how this tracing happens.  In HttpConnection, we have the Trace method:
https://github.com/dotnet/corefx/blob/338f96aee6232f3dc2a47af51f839c48d93a8a22/src/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/HttpConnection.cs#L1603-L1609
which tries to provide all of the relevant info about the request, as shown by the comments on those arguments.  We end up missing that when going to these other NetEventSource methods directly.

Reply:
>'m a little concerned we're starting to become inconsistent (or maybe we already were and this just makes it more so)

We are inconsistent in a lot of place.  One area that I would like to see improved is for us to use "error" vs. "info" better.

We have NetEventSource.Info() vs. NetEventSource.Error() which is easy to find errors since they tagged with the right event id level (verbose vs. error).

The newly added "HandlerMessage" event id seems just for "info" things.  If we were to start using that, we need to split it up perhaps into 2 different ids (HandlerMessageInfo vs. HandlerMessageError) or perhaps use the same event id but use different EventLevel depending on which tracing method we call.

We need to get better so that finding "errors" is easier and filter out from all the other verbose output.

Reply:
Sure.  My point, though, is that in SocketsHttpHandler, that is the primary event that's used for these kind of info messages:
https://source.dot.net/#System.Net.Http/System/Net/Http/SocketsHttpHandler/HttpConnection.cs,c38d0442100389ce,references
If we're going through and adding more logging, let's try to do so in a consistent fashion moving forward, rather than creating more work to be cleaned up later.

Reply:
I'll look into what we can do to keep this consistent with the existing logging and can get back to you on this.

Reply:
This looks like this should be relatively straightforward, but I have run into one thing I'd like explain before I make the change. In `HttpConnection.Trace`, the first parameter is the hash code of the `HttpConnectionPool`, aka pool ID
https://github.com/dotnet/corefx/blob/338f96aee6232f3dc2a47af51f839c48d93a8a22/src/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/HttpConnection.cs#L1603-L1609

However, based on the trace string and the argument name, it looks like that parameter was intended to be the hash code of the handler, and not the pool:
https://github.com/dotnet/corefx/blob/338f96aee6232f3dc2a47af51f839c48d93a8a22/src/System.Net.Http/src/System/Net/Http/NetEventSource.Http.cs#L60-L63
<img width="455" alt="untitled" src="https://user-images.githubusercontent.com/4656175/39828842-50120b16-5371-11e8-94fe-f10336f2faae.png">

To fix this inconsistency I'll rename the parameter to make it clear that it's actually the ID of the HttpConnectionPool that is being logged. I'll also verify that we're logging the associations between HttpConnectionPool and SocketsHttpHandler.

Other than that, the updated change will essentially just switch over the RedirectHandler to use a similar model to `HttpConnection` by adding a trace method that calls `NetEventSource.Log.HandlerMessage`.

Reply:
I would prefer we change how those hash codes are being displayed in the traces.  They show up right now as decimal (with commas separating the digits).  They should simply be done as hex, i.e. `ToString("X")`

Reply:
They're output as numbers; it's PerfView that's rendering them from the traced data that way.

Reply:
>They're output as numbers

Maybe they should be output as strings instead, i.e. with "ToString("X") and prepended with "0x".  It would make them more readable.

Reply:
That will force every tool to see them as a string and limit how they can be rendered.  PerfView could easily be updated to have an option to show numbers a hex or decimal, if it doesn't already. But tracing the data as a string would effectively prevent that.  It's also more expensive to trace as a string.

Reply:
>It's also more expensive to trace as a string

Ok.  I'll submit a issue/PR to the PerfView repo.

Reply:
@davidsh I agree on the error traces! Adding that should be very straightforward. For displaying IDs in hex I think that fixing it in perfview is the right approach. It will definitely make things more readable, and should probably be an easy fix.